File: otona_book_flutter/lib/common/api_client.dart
import 'dart:io';

import 'package:dio/dio.dart';
import 'package:dio/io.dart';
import 'package:flutter/material.dart';
import 'package:otona_book_flutter/common/global.dart';
import 'package:otona_book_flutter/models/api_dto.dart';

class ApiClient {
  ApiClient([this.context]) {
    _options = Options(extra: {"context": context});
  }

  BuildContext? context;
  late Options _options;
  static Dio dio = Dio(BaseOptions(
      baseUrl: "http://localhost:7129",
      headers: {HttpHeaders.acceptHeader: "application/json"}));

  Future<dynamic> post(String route, Object? data) async {
    var response = await dio.post<dynamic>(route, data: data);
    var responseData = ApiResponse.fromJson(response.data);

    if (responseData.code != 0) {
      throw Error();
    }

    return responseData.data;
  }

  static void init() {
    if (!Global.isRelease) {
      (dio.httpClientAdapter as DefaultHttpClientAdapter).onHttpClientCreate =
          (client) {
        // client.findProxy = (uri) {
        //   return 'PROXY 192.168.50.154:8888';
        // };
        //代理工具会提供一个抓包的自签名证书，会通不过证书校验，所以我们禁用证书校验
        client.badCertificateCallback =
            (X509Certificate cert, String host, int port) => true;
      };
    }
  }
}

File: otona_book_collector/src/jobs/avmoo/api-client.js
const axios = require('axios');

const baseUrl = 'http://localhost:7129';

const post = async (route, data, options) => {
  const response = await axios.post(`${baseUrl}${route}`, data);
  const { data: response_data } = response;

  if (response_data) {
    const { code, data, msg, sub_code, sub_msg } = response_data;
    if (code) {
      let error_msg = msg;
      if (sub_code && sub_msg) {
        error_msg += ` ### [${sub_code}] ${sub_msg}`;
      }
      throw new Error(error_msg);
    }

    return data;
  } else {
    throw new Error('NETWORK ERROR');
  }
};

module.exports = { post };
